pkg/datapath/loader/base.go

```diff
@@ -408,6 +410,17 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 					}
 					interfaces = append(interfaces, iface)
 				}
+				// Discover any other devices that might be needed. Here we know EKS
+				// names its interfaces eth0, eth1, eth2, so we pattern match this
+				// case.
+				if links, err := netlink.LinkList(); err == nil {
+					for _, link := range links {
+						linkName := link.Attrs().Name
+						if linkName != iface && strings.Contains(linkName, "eth*") {
```

Comment:
`eth` is distro-specific, many distros use the new style of device naming scheme based on physical system layouts (en*) 

We hit an issue like that and resolved in https://github.com/cilium/cilium/pull/13234 , but in that case we were only cocerned with the device with the default route, not all devices.

Reply:
The other logic I considered is to use driver attached to the device, but that would also likely require cases for eks, ubuntu, etc.

Reply:
Or we could learn it from the source routes I guess which is probably the most reliable.

Reply:
Any particular reason we can't make this consistent with the interface list detection introduced in the earlier commits of this tree? Just that it's trying to pull the "latest" set of all interfaces at Reinitialize time in case the device list changes during Cilium operation?

Reply:
FWIW since I understand this is "additive", ie we try to append a set of interfaces to a list that already exists and could otherwise be populated by the user, I don't think this is necessarily a blocker for fixing this... but at the same time, if we're taking the effort to simplify these steps for other distros, then it makes sense for us to also follow up on Ubuntu since that's one of the most common distros used in cloud.

Last time we made an assumption about `eth0`, we found out within a week, maybe even same day that it was a problem.

Reply:
Discussed out-of-band, should be good enough to have an exclusion list for virtual device types (veth, ipvlan, tunnel?) and select any other devices.

Reply:
Updates here, https://github.com/cilium/cilium/pull/15357/commits/3deb07c7a556e9b7accb24fafe428daf0d4ee90e